include/ignition/physics/detail/SpecifyData.hh

```diff
@@ -39,6 +39,23 @@ namespace ignition
       // Do nothing
     }
 
+    /////////////////////////////////////////////////
+    template <typename Expected>
+    ExpectData<Expected>::ExpectData(const ExpectData<Expected> &_other)
+        : ExpectData()
+    {
+      this->Copy(_other);
```

Comment:
It's been a long time since I've looked at this code, but I don't think this is the "right" thing to do here, even if it gives the correct outcome. If I remember how this all works, this will cause all of the data in `CompositeData` to be re-copied over and over again, once for each data type `T` that was given to an `ExpectData<T>`.

I think we can just leave out this line. I believe delegating to the default constructor should be enough to fix the problem described by the PR.

Reply:
Yeah, you're right. The copy constructor of `CompositeData` is called before this and copies the `MapData` object so all that is left to do here is point the `PrivateSpecifyData::expectedIterator` to the new map. https://github.com/ignitionrobotics/ign-physics/pull/77/commits/ec3947b318120951011ec1982c1c81f8abd63ffd